op-acceptance-tests/tests/interop/loadtest/interop_load_test.go

```diff
@@ -1,139 +1,109 @@
-	initMsgsTx := txintent.NewIntent[*txintent.MultiTrigger, *txintent.InteropOutput](eoaA.Plan())
-	initCalls := make([]txintent.Call, 0, numMsgs)
-	rng := rand.New(rand.NewSource(1234))
-	for range numMsgs {
-		initCalls = append(initCalls, interop.RandomInitTrigger(rng, eventLoggerAddress, rng.Intn(5), rng.Intn(10)))
+	l2ELA := sys.L2ChainA.PublicRPC()
+	L2A := &L2{
+		Chain:  sys.L2ChainA,
+		EL:     l2ELA,
+		Funder: dsl.NewFunder(sys.Wallet, sys.FaucetA, l2ELA),
 	}
-	initMsgsTx.Content.Set(&txintent.MultiTrigger{
-		Emitter: constants.MultiCall3,
-		Calls:   initCalls,
-	})
-	initResult, err := initMsgsTx.Result.Eval(t.Ctx())
-	t.Require().NoError(err)
-	t.Require().Len(initResult.Entries, numMsgs)
-	_, err = initMsgsTx.PlannedTx.Success.Eval(t.Ctx())
-	t.Require().NoError(err)
-
-	// Wait to include the exec txs until we know it will be included at a higher timestamp than initMsgsTx, modulo reorgs.
-	// NOTE: this should be `<`, but the mempool filtering in op-geth currently uses the unsafe head's timestamp instead of
-	// the pending timestamp. See https://github.com/ethereum-optimism/op-geth/issues/603.
-	for l2ELA.BlockRefByLabel(eth.Unsafe).Time <= initMsgsTx.PlannedTx.IncludedBlock.Value().Time {
-		l2ELA.WaitForBlock()
+	l2ELB := sys.L2ChainB.PublicRPC()
+	L2B := &L2{
+		Chain:  sys.L2ChainB,
+		EL:     l2ELB,
+		Funder: dsl.NewFunder(sys.Wallet, sys.FaucetB, l2ELB),
 	}
 
-	execCalls := make([]txintent.Call, 0, numMsgs)
-	for i := range numMsgs {
-		execCalls = append(execCalls, &txintent.ExecTrigger{
-			Executor: constants.CrossL2Inbox,
-			Msg:      initResult.Entries[i],
-		})
-	}
-	const numExecTxs = 1_000
-	execMsgsTxs := make([]*txintent.IntentTx[*txintent.MultiTrigger, txintent.Result], 0, numExecTxs)
-	for i := range numExecTxs {
-		execMsgsTx := txintent.NewIntent[*txintent.MultiTrigger, txintent.Result](eoaB.Plan(), txplan.WithRetryInclusion(l2ELB.Escape().EthClient(), 50, retry.Exponential()))
-		execMsgsTx.Content.Set(&txintent.MultiTrigger{
-			Emitter: constants.MultiCall3,
-			Calls:   execCalls,
-		})
-		if i != 0 {
-			prevNonce := &execMsgsTxs[i-1].PlannedTx.Nonce
-			execMsgsTx.PlannedTx.Nonce.DependOn(prevNonce)
-			execMsgsTx.PlannedTx.Nonce.Fn(func(_ context.Context) (uint64, error) {
-				prevNonceU64, err := prevNonce.Get()
-				if err != nil {
-					return 0, err
-				}
-				return prevNonceU64 + 1, nil
-			})
-		}
-		execMsgsTxs = append(execMsgsTxs, execMsgsTx)
-	}
 	var wg sync.WaitGroup
-	wg.Add(len(execMsgsTxs))
-	for _, execMsgsTx := range execMsgsTxs {
-		go func(tx *txplan.PlannedTx) {
-			defer wg.Done()
-			receipt, err := tx.Included.Eval(t.Ctx())
-			t.Require().NoError(err)
-			t.Require().Len(receipt.Logs, numMsgs)
-			_, err = tx.Success.Eval(t.Ctx())
-			t.Require().NoError(err)
+	defer wg.Wait()
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		SpamInteropTxs(t, &wg, numInitTxs, L2A, L2B, sys.Supervisor)
```

Comment:
Rather than using one wg and passing it in here it would be cleaner if SpamInteropTxs were to create its own and block until that wg had finished.

Reply:
I'll implement this.

For my own education: what is the reasoning behind this being cleaner? Passing it in allows the goroutines to end a bit sooner (releasing some resources in the process).